include/fifo_cache_policy.hpp

```diff
@@ -49,7 +51,9 @@ class FIFOCachePolicy : public ICachePolicy<Key>
     // handle element deletion from a cache
     void Erase(const Key &key) noexcept override
     {
-        fifo_queue.pop_back();
+        auto it = key_lookup.find(key);
```

Comment:
Policy's `Erase` called only if an element exists in a cache so that must exist in a policy. I'd suggest the following changes, as calling `find()` without checking for element presence (`it != key_lookup.cend()`) does not make a lot of sense in my opinion:

```suggestion
        auto element = key_lookup[key];
        fifo_queue.erase(it->second);
        key_lookup.erase(it->first);
```